src/dmd/dmangle.d

```diff
@@ -33,13 +33,38 @@ import dmd.identifier;
 import dmd.mtype;
 import dmd.root.ctfloat;
 import dmd.root.outbuffer;
+import dmd.root.stringtable;
 import dmd.root.aav;
 import dmd.root.string;
 import dmd.target;
 import dmd.tokens;
 import dmd.utf;
 import dmd.visitor;
 
+
+/**********************************************
+ * Convert a string representing a type (the deco) and
+ * return its equivalent Type.
+ * Params:
+ *      deco = string containing the deco
+ * Returns:
+ *      null for failed to convert
+ *      Type for succeeded
+ */
+
+public Type decoToType(const(char)[] deco)
+{
+    auto sv = Type.stringtable.lookup(deco);
```

Comment:
I think this is way too simplistic: you cannot synthesize new types and you cannot create types that the compiler hasn't yet seen, but are in other modules, structs or similar.

I don't yet see how `__totype` does anything more than optimize `mixin(core.demangle(deco))` (but allowing access to private types).

Reply:
Ah, so that only works with strings that were previously returned by `.mangleof`. Yes that may be problematic.